packages/walletconnect-solana/src/core.ts

```diff
@@ -148,67 +158,101 @@ export class WalletConnectWallet {
-					},
+		const { signature, transaction: signedSerializedTransaction } = await this.client.client.request<{
+			signature: string
+			transaction?: string
+		}>({
+			chainId: this._network,
+			topic: this.session.topic,
+			request: {
+				method: WalletConnectRPCMethods.signTransaction,
+				params: {
+					// Passing ...legacyTransaction is deprecated.
+					// All new clients should rely on the `transaction` parameter.
+					// The future versions will stop passing ...legacyTransaction.
+					...legacyTransaction,
+					// New base64-encoded serialized transaction request parameter
+					transaction: this.serialize(transaction),
 				},
-			})
-			transaction.addSignature(this.publicKey, Buffer.from(base58.decode(signature)))
+			},
+		})
 
-			return transaction
-		} else {
-			throw new ClientNotInitializedError()
+		if (signedSerializedTransaction) {
+			return this.deserialize(signedSerializedTransaction, isVersioned) as T
 		}
+
+		transaction.addSignature(this.publicKey, Buffer.from(base58.decode(signature)))
+
+		return transaction
 	}
 
 	async signMessage(message: Uint8Array): Promise<Uint8Array> {
-		if (this._UniversalProvider?.session && this._session) {
-			const { signature } = await this._UniversalProvider.client.request<{ signature: string }>({
-				// The network does not change the output of message signing, but this is a required parameter for SignClient
-				chainId: this._network,
-				topic: this._session.topic,
-				request: {
-					method: WalletConnectRPCMethods.signMessage,
-					params: { pubkey: this.publicKey.toString(), message: base58.encode(message) },
+		this.checkIfWalletSupportsMethod(WalletConnectRPCMethods.signMessage)
+
+		const { signature } = await this.client.client.request<{
+			signature: string
+		}>({
+			// The network does not change the output of message signing, but this is a required parameter for SignClient
+			chainId: this._network,
+			topic: this.session.topic,
+			request: {
+				method: WalletConnectRPCMethods.signMessage,
+				params: {
+					pubkey: this.publicKey.toString(),
+					message: base58.encode(message),
 				},
-			})
+			},
+		})
 
-			return base58.decode(signature)
-		} else {
-			throw new ClientNotInitializedError()
-		}
+		return base58.decode(signature)
+	}
+
+	async signAndSendTransaction<T extends Transaction | VersionedTransaction>(transaction: T): Promise<string> {
+		this.checkIfWalletSupportsMethod(WalletConnectRPCMethods.signAndSendTransaction)
+
+		const { signature } = await this.client.client.request<{
+			signature: string
+		}>({
+			chainId: this._network,
+			topic: this.session.topic,
+			request: {
+				method: WalletConnectRPCMethods.signAndSendTransaction,
+				params: { transaction: this.serialize(transaction) },
+			},
+		})
+
+		return signature
+	}
+
+	async signAllTransactions<T extends Transaction | VersionedTransaction>(transactions: T[]): Promise<T[]> {
+		this.checkIfWalletSupportsMethod(WalletConnectRPCMethods.signAllTransactions)
```

Comment:
how can the dapp know before hand if the wallet supports these methods?

Reply:
It is present in the namespace we receive when there is a successful connection. We can see [here](https://github.com/WalletConnect/walletconnect-solana-adapter/pull/42/files#diff-4ef08e8776fc9a3ab06dd2bb2c371ec838f9e6c76b07f54eb426b73f8935f44cR294) the logic.

Reply:
I mean the dev using the adapter

Reply:
I'm not sure if I understood the question, but he could get it in the session or catch the error